recipes/utils/p7zip.yaml

```
@@ -0,0 +1,36 @@
+inherit: [cpackage, make, install]
+
+metaEnvironment:
+    PKG_VERSION: "17.05"
+    PKG_LICENSE: "LGPL-2.1+ with unRAR restriction"
+
+checkoutSCM:
+    scm: url
+    url: ${GITHUB_MIRROR}/p7zip-project/p7zip/archive/refs/tags/v${PKG_VERSION}.tar.gz
+    digestSHA1: f8c03d6b55759867fb464bfe5caf49149de34b01
+    stripComponents: 1
+
+buildVars: [CC, CXX, CFLAGS, CPPFLAGS, LDFLAGS]
+buildScript: |
+    mkdir -p build
+    pushd build
+    cp -arfs $1/* .
+    makeParallel \
+        CC="$CC" ALLFLAGS_C="$CFLAGS" \
+        CXX="$CXX" ALLFLAGS_CPP="$CPPFLAGS" \
+        LDFLAGS="$LDFLAGS" \
+        7z 7za 7zr
+
+    makeParallel \
+        DEST_HOME=/usr \
+        DEST_DIR=$BOB_CWD/install \
+        install
+    popd
+
+packageScript: |
+    installCopy "$1/install/"
```

```suggestion
    installCopy "$1/install/"

     pushd usr/bin
     cat > 7z << 'EOF'
     #!/bin/bash
     exec $(dirname ${BASH_SOURCE[0]})/../local/lib/p7zip/7z $@
     EOF
     chmod +x 7z
     popd #usr/bin

```

Maybe also for  `7za`, `7zr` ?

---

Thanks. The `local` directory looks suspicious. That should not be used...

---

> Maybe also for `7za`, `7zr` ?

Thanks for the input. I made a fix which does something similar. So this works now without a user has to specify the full path.